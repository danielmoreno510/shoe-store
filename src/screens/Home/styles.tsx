import {StyleSheet} from 'react-native';
import {widthPercentageToDP} from '../../utils/responsiveUtils';

export default StyleSheet.create({
  container: {
    flex: 1,
    padding: widthPercentageToDP('5%'),
    backgroundColor: 'white',
  },
  title: {
    fontSize: 28,
    fontWeight: 'bold',
  },
  text: {
    paddingTop: 8,
    fontSize: 18,
    textAlign: 'justify',
  },
  button: {
    marginTop: 40,
  },
});
