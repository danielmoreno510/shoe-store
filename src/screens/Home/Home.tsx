import React from 'react';
import {ScrollView, View, Text} from 'react-native';

import Button from '../../components/common/Button';
import styles from './styles';

const Home: React.FC<Navigation> = ({navigation}) => (
  <ScrollView style={styles.container}>
    <Text style={styles.title}>Sneaker City</Text>
    <Text style={styles.text}>Welcome to Sneaker City, enjoy our shoes</Text>
    <View style={styles.button}>
      <Button text="Shoes" onPress={() => navigation.navigate('Shoes List')} />
    </View>
  </ScrollView>
);

export default Home;
