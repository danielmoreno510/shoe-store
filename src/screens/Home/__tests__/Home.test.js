import React from 'react';
import {shallow} from 'enzyme';

import Home from '../Home';

describe('Home component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      navigation: {navigate: jest.fn()},
    };

    describe('WHEN the component is rendered', () => {
      const shallowWrapper = shallow(<Home {...testProps} />);

      it('THEN should display a regular Home', () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });

    describe('WHEN the shoes button is pressed', () => {
      const shallowWrapper = shallow(<Home {...testProps} />);
      shallowWrapper.find('CustomButton').simulate('press');

      it('THEN navigation function should be called', () => {
        expect(testProps.navigation.navigate).toBeCalledTimes(1);
      });
    });
  });
});
