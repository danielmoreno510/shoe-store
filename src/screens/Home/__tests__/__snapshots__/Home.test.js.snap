// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Home 1`] = `
<ScrollView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 37.5,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 28,
        "fontWeight": "bold",
      }
    }
  >
    Sneaker City
  </Text>
  <Text
    style={
      Object {
        "fontSize": 18,
        "paddingTop": 8,
        "textAlign": "justify",
      }
    }
  >
    Welcome to Sneaker City, enjoy our shoes
  </Text>
  <View
    style={
      Object {
        "marginTop": 40,
      }
    }
  >
    <CustomButton
      onPress={[Function]}
      text="Shoes"
    />
  </View>
</ScrollView>
`;
